src/jsonschema.rs

```diff
@@ -68,17 +75,91 @@ struct Tag {
     extra: HashMap<String, Value>,
 }
 
-struct JSONSchema {
-    data: Tag,
-}
+impl Tag {
+    fn get_type(&self) -> Type {
+        match &self.data_type {
+            Value::String(string) => {
+                let atom: Atom = serde_json::from_value(json!(string)).unwrap();
+                Type::Atom(atom)
+            }
+            Value::Array(array) => {
+                let list: Vec<Atom> = array
+                    .iter()
+                    .map(|v| serde_json::from_value(json!(v)).unwrap())
+                    .collect();
+                Type::List(list)
+            }
+            _ => Type::Atom(Atom::Object),
+        }
+    }
+
+    pub fn type_into_ast(&self) -> ast::Tag {
+        match self.get_type() {
+            Type::Atom(atom) => self.atom_into_ast(atom),
+            Type::List(list) => {
+                let mut items: Vec<ast::Tag> = Vec::new();
+                for atom in list {
+                    items.push(self.atom_into_ast(atom));
+                }
+                ast::Tag::new(ast::Type::Union(ast::Union::new(items)), None, false)
+            }
+        }
+    }
+
+    fn atom_into_ast(&self, data_type: Atom) -> ast::Tag {
+        match data_type {
+            Atom::Null => ast::Tag::new(ast::Type::Null, None, false),
+            Atom::Boolean => ast::Tag::new(ast::Type::Atom(ast::Atom::Boolean), None, false),
+            Atom::Number => ast::Tag::new(ast::Type::Atom(ast::Atom::Number), None, false),
+            Atom::Integer => ast::Tag::new(ast::Type::Atom(ast::Atom::Integer), None, false),
+            Atom::String => ast::Tag::new(ast::Type::Atom(ast::Atom::String), None, false),
+            Atom::Object => match &self.object.properties {
```

Comment:
This pattern matching logic is ugly, but is necessary because emulated types like maps are embedded within objects.